nni/common/concrete_trace_utils/concrete_tracer.py

```diff
@@ -1376,12 +1405,12 @@ def f(x, y):
         else:
             assert node_a.op == node_b.op and target_a == target_b
 
-    with MagicMethodPatcher(tracer.root):
+    with MagicMethodPatcher():
         name = root.__class__.__name__ if isinstance(root, torch.nn.Module) else root.__name__
         traced = GraphModule(tracer.root, graph, name)
 
     # TODO: better infomation
     # # assert root(**concrete_args) == traced(**concrete_args)
     if check_args is not None:
-         assert root(**check_args) == traced(**check_args)
-    return traced
+        assert root(**check_args) == traced(**check_args)
```

Comment:
do not return tracer

Reply:
yes, why return the tracer?

Reply:
Here returns the tracer because the tracer contains module path information needed. Reference the implementation of https://github.com/pytorch/examples/blob/main/fx/module_tracer.py.

Reply:
The latest commit has dealt with this issue by storing the module path information into the returned GraphModule, i.e., the `traced`. So returning the tracer can be avoided.